source/extensions/filters/network/thrift_proxy/conn_manager.cc

```diff
@@ -201,13 +201,29 @@ FilterStatus ConnectionManager::ResponseDecoder::fieldBegin(absl::string_view na
     // Reply messages contain a struct where field 0 is the call result and fields 1+ are
     // exceptions, if defined. At most one field may be set. Therefore, the very first field we
     // encounter in a reply is either field 0 (success) or not (IDL exception returned).
-    success_ = field_id == 0 && field_type != FieldType::Stop;
+    // If first fieldType is FieldType::Stop then it is a void success and handled in messageEnd()
+    // because decoder state machine does not call decoder event callback fieldBegin on FieldType::Stop.
+    success_ = field_id == 0;
```

Comment:
@zuercher should I revert the change on this line? Given the comment I added the existing code was confusing.